### DISCUSSION TITLE
kableExtra to svg for word export

### DISCUSSION THREAD

**Discussion by tomas-brito-devoto** [haozhu233/kableExtra#732] [Ideas]
Hi!
I'm writing my thesis with quarto, and I've been wanting to render my kableExtra tables into word. I know I can copy&paste the html generated tables from my browser into word, but that isn't an automatic solution when rendering reports. Ideally, I would like to render tables directly into word, posibbly having my table genetaring code in a separate script and importing the table into the manuscript markdown/quarto file. (This could be acomplished with `xml2::read_html` and  `xml_as_kable` functions.) 

Another solution I thought about is exporting the tables in svg format and importing the table as an image. Since kableExtra already exports to html and works with xml files, maybe it could be possible? I've been looking a lot on the internet but I found no solution yet. 

I did find some HTML to SVG/pdf converters on this [stackoverflow post](https://stackoverflow.com/questions/12005860/can-i-export-part-of-an-html-page-to-an-svg-image/26963688#26963688). I haven't tested them yet, but, again, it is not an automatic process. 

Thanks in advance for an awesome package!
T.

**Comment by tomas-brito-devoto** [comment]
Ok, I've tried the HTML to SVG converters and they did not work for me. In some cases some empty svg files were created, and for other programs, dependencies were too complicated for me to follow.

**Comment by haozhu233** [comment]
There is a to_image function which does a similar job. Although It's depending on phantomjs which is sort of deprecated. 

https://haozhu233.github.io/kableExtra/save_kable_and_as_image.html

**Comment by tomas-brito-devoto** [comment]
Thank you for your promptly reply @haozhu233!
Indeed, saving the table an png/jpg image would allow me to automatically insert it into my manuscript. Unfortunatelly, `save_kable` and `as_image` don't support svg format, so there is some quality loss. It will be my current workflow until I figure out a better alternative. (Rendering pdf tables into word is not possible).
